membership.go

```diff
@@ -284,6 +285,30 @@ func guessMulticastAddress() string {
 	return multicastAddress
 }
 
+func getListenInterface() (*net.Interface, error) {
```

Comment:
Would you mind just adding a little commentary around this function to describe its rationale and how it's doing what it does?

Reply:
Fair enough, I will update the PR tomorrow.